### PR TITLE
Python: add ZLEXCOUNT command

### PR DESCRIPTION
**Pull request by aaron-congo**
https://redis.io/docs/latest/commands/zlexcount/

### CODE REVIEW

File: python/python/glide/async_commands/core.py [hunk 1]
```diff
@@ -1986,6 +1987,49 @@ async def zremrangebyscore(
...
+        max_lex: Union[InfBound, LexBoundary],
+    ) -> int:
+        """
+        Returns the number of members in the sorted set stored at `key` with scores between `min_lex` and `max_lex`.
+
+        See https://redis.io/commands/zlexcount/ for more details.
+
+        Args:
+            key (str): The key of the sorted set.
+            min_lex (Union[InfBound, LexBoundary]): The minimum lex to count from.
```
Comment (SanHalacogluImproving) [inline python/python/glide/async_commands/core.py:2003]: could you change lex = lexicographical order or lexicographic (where or whatever you deem to be appropriate lex doesn't seem to be meaningful). I think it makes more sense. After you have, I'll update the java docs. What do you think?

File: python/python/glide/async_commands/transaction.py [hunk 2]
```diff
@@ -1524,6 +1525,40 @@ def zremrangebyscore(
...
+        min_lex: Union[InfBound, LexBoundary],
+        max_lex: Union[InfBound, LexBoundary],
+    ) -> TTransaction:
+        """
+        Returns the number of members in the sorted set stored at `key` with scores between `min_lex` and `max_lex`.
+
+        See https://redis.io/commands/zlexcount/ for more details.
+
+        Args:
+            key (str): The key of the sorted set.
```
Comment (SanHalacogluImproving) [inline python/python/glide/async_commands/transaction.py:1540]: Same comment as above.

File: python/python/tests/test_async_client.py [hunk 3]
```diff
@@ -1350,6 +1350,49 @@ async def test_zremrangebyscore(self, redis_client: TRedisClient):
...
+            )
+            == 2
+        )
+        assert (
+            await redis_client.zlexcount(
+                key1, InfBound.NEG_INF, LexBoundary("c", is_inclusive=True)
+            )
+            == 3
+        )
+        # Incorrect range; start > end
```
Comment (Yury-Fridlyand) [inline python/python/tests/test_async_client.py:1378]: move comment to the next test

File: python/python/tests/test_async_client.py [hunk 4]
```diff
@@ -1350,6 +1350,49 @@ async def test_zremrangebyscore(self, redis_client: TRedisClient):
...
+            == 0
+        )
+        assert (
+            await redis_client.zlexcount(
+                "non_existing_key", InfBound.NEG_INF, InfBound.POS_INF
+            )
+            == 0
+        )
+
+        assert await redis_client.set(key2, "value") == OK
```
Comment (Yury-Fridlyand) [inline python/python/tests/test_async_client.py:1392]: comment

**Review by Yury-Fridlyand — APPROVED** [review]

File: python/python/glide/async_commands/core.py (hunk 1 above)
Comment (aaron-congo) [reply python/python/glide/async_commands/core.py]: sure. I used lexicographical because that was used elsewhere in the file.

**Pull request merged**